src/wallet/walletdb.cpp

```diff
@@ -998,11 +999,11 @@ DBErrors WalletBatch::FindWalletTx(std::vector<uint256>& vTxHash, std::list<CWal
             CDataStream ssKey(SER_DISK, CLIENT_VERSION);
             CDataStream ssValue(SER_DISK, CLIENT_VERSION);
             bool complete;
-            bool ret = m_batch->ReadAtCursor(ssKey, ssValue, complete);
+            bool ret = cursor->Next(ssKey, ssValue, complete);
             if (complete) {
                 break;
             } else if (!ret) {
-                m_batch->CloseCursor();
+                cursor.reset();
```

Comment:
In f2c83f4b:

nit: as the cursor will be cleaned on the destructor, and we are returning here, this `reset` call is not needed.

Reply:
Done